cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperatorTest.java

```diff
@@ -0,0 +1,120 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockserver.integration.ClientAndServer;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(VertxExtension.class)
+public class KafkaClusterRebalanceAssemblyOperatorTest {
+
+    private static final String HOST = "localhost";
+    private static final String CLUSTER_NAMESPACE = "cruise-control-namespace";
+    private static final String CLUSTER_NAME = "cruise-control-test-cluster";
+
+    private final KubernetesVersion kubernetesVersion = KubernetesVersion.V1_11;
+
+    private static final Logger log = LogManager.getLogger(KafkaClusterRebalanceAssemblyOperatorTest.class.getName());
+
+    private static ClientAndServer ccServer;
+
+    @BeforeAll
+    public static void before() throws IOException, URISyntaxException {
+        ccServer = MockCruiseControl.getCCServer(CruiseControl.REST_API_PORT, 0);
+    }
+
+    @AfterAll
+    public static void after() {
+        ccServer.stop();
+    }
+
+    @Test
+    public void testNewRebalance(Vertx vertx, VertxTestContext context) {
+
+        KafkaClusterRebalanceBuilder kcrBuilder = new KafkaClusterRebalanceBuilder();
+
+        Map<String, String> labels = new HashMap<>();
+        labels.put(Labels.STRIMZI_CLUSTER_LABEL, "my-test-cluster");
+        ObjectMeta meta = new ObjectMetaBuilder().withLabels(labels).withName(CLUSTER_NAME).withNamespace(CLUSTER_NAMESPACE).build();
+
+        KafkaClusterRebalanceSpec rebalanceSpec = new KafkaClusterRebalanceSpecBuilder().build();
+
+        Condition newRebalanceCondition = new Condition();
+        newRebalanceCondition.setType(String.valueOf(KafkaClusterRebalanceAssemblyOperator.State.New));
+
+        KafkaClusterRebalance kcRebalance = kcrBuilder
+                .withMetadata(meta)
+                .withSpec(rebalanceSpec)
+                .withNewStatus()
+                    .withConditions(newRebalanceCondition)
+                .endStatus()
+                .build();
+
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(true);
+        PlatformFeaturesAvailability pfa = new PlatformFeaturesAvailability(true, kubernetesVersion);
+        KafkaClusterRebalanceAssemblyOperator kcrao = new KafkaClusterRebalanceAssemblyOperator(vertx, pfa, supplier, HOST);
+
+        CrdOperator<KubernetesClient,
+                KafkaClusterRebalance,
+                KafkaClusterRebalanceList,
+                DoneableKafkaClusterRebalance> mockRebalanceOps = supplier.kafkaClusterRebalanceOperator;
+
+        when(mockRebalanceOps.get(CLUSTER_NAMESPACE, CLUSTER_NAME)).thenReturn(kcRebalance);
+        when(mockRebalanceOps.getAsync(anyString(), anyString())).thenReturn(Future.succeededFuture(kcRebalance));
```

Comment:
the problem I see mocking in this way, is that we are returning always the initial `kcRebalance` instance while during the reconcile, in the real world, its status is updated with proposal information, session-id, etc etc.
I think it's our first time doing a mock test where the mocked class has to return an updated instance "at some point".

Reply:
Yeah this is the part where I wanted your input on how to proceed.

Reply:
Not a mockito expert so I need to think/study more about that :-D